*** ldmain.c    2014-10-14 07:32:04.000000000 +0000
--- ldmain.modify.c     2015-05-14 16:24:47.264961161 +0000
***************
*** 579,584 ****
--- 579,586 ----
  /* We need to find any explicitly given emulation in order to initialize the
     state that's needed by the lex&yacc argument parser (parse_args).  */

+ const char * mips_emul_mode = "elf32elmip";
+
  static char *
  get_emulation (int argc, char **argv)
  {
***************
*** 611,619 ****
--- 613,623 ----
                   || strcmp (argv[i], "-mips5") == 0
                   || strcmp (argv[i], "-mips32") == 0
                   || strcmp (argv[i], "-mips32r2") == 0
+                  || strcmp (argv[i], "-mips32r5") == 0
                   || strcmp (argv[i], "-mips32r6") == 0
                   || strcmp (argv[i], "-mips64") == 0
                   || strcmp (argv[i], "-mips64r2") == 0
+                  || strcmp (argv[i], "-mips64r5") == 0
                   || strcmp (argv[i], "-mips64r6") == 0)
            {
              /* FIXME: The arguments -mips1, -mips2, -mips3, etc. are
***************
*** 623,628 ****
--- 627,633 ----
                 implemented as emulations; until then, we just ignore
                 the arguments and hope that nobody ever creates
                 emulations named ips1, ips2 or ips3.  */
+             emulation = mips_emul_mode;
            }
          else if (strcmp (argv[i], "-m486") == 0)
            {
